sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -34,22 +34,184 @@ import org.apache.spark.sql.rapids.RapidsDiskBlockManager
  */
 class DuplicateBufferException(s: String) extends RuntimeException(s) {}
 
+/**
+ * An object that client code uses to interact with an underlying RapidsBufferId.
+ *
+ * A handle is obtained when a buffer, batch, or table is added to the spill framework
+ * via the `RapidsBufferCatalog` api.
+ */
+trait RapidsBufferHandle {
+  val id: RapidsBufferId
```

Comment:
Perhaps I am missing something here. Why is this a trait and not just a class with the ID hidden? I thought we wanted the handle to be opaque and exposing the ID is not opaque. It looks like the only place that does not use RapidsBufferCatalog. RapidsBufferHandleImpl is one test file. I would prefer to update the test and make this opaque if possible.

Reply:
I can make that change. I was trying to restrain the amount of change, but this is not going to be too much work. 

The id is being used from any of the catalogs right now: RapidsBufferCatalog, ShuffleBufferCatalog, and ShuffleReceivedBufferCatalog (in addition to the tests you mentioned). We tend to use id in these catalogs to store stuff about the RapidsBuffers, but I can try to make all of them use the handle. 

The UCX shuffle breaks the ids a bit. The shuffle catalogs creates `RapidsBufferId` instances that have the `ShuffleBlockId` in them. We could have this id be external to the store (i.e. the store hands you back a `RapidsBufferId` and then the shuffle is going to map that to the `ShuffleBlockId`). Did I understand correctly? Do we want the change in this PR or subsequent ones?

Reply:
On the trait: I was trying to use the trait as the public interface to `RapidsBufferHandle` (id could be removed from that, but I think I need to move a couple of classes around to a different package if want to hide it all together)

The impl knows about the catalog, and is defined inside of `RapidsBufferCatalog`.

Reply:
This is minor. If it is going to be hard or we have to move a lot around we can just go with a comment.